src/coinbase/wallet.ts

```diff
@@ -289,6 +293,121 @@ export class Wallet {
+   * @returns The unstakeable balance.
+   */
+  public async unstakeableBalance(
+    asset_id: string,
+    mode: StakeOptionsMode = StakeOptionsMode.DEFAULT,
+    options: { [key: string]: string } = {},
+  ): Promise<Decimal> {
+    if (!this.getDefaultAddress()) {
+      throw new InternalError("Default address not found");
+    }
+    return await this.getDefaultAddress()!.unstakeableBalance(asset_id, mode, options);
+  }
+
+  /**
+   * Get the claimable balance for the supplied asset.
+   *
+   * @param asset_id - The asset to check claimable balance for.
+   * @param mode - The staking mode. Defaults to DEFAULT.
+   * @param options - Additional options for getting the claimable balance.
+   * @throws {Error} if the default address is not found.
+   * @returns The claimable balance.
+   */
+  public async claimableBalance(
+    asset_id: string,
+    mode: StakeOptionsMode = StakeOptionsMode.DEFAULT,
+    options: { [key: string]: string } = {},
+  ): Promise<Decimal> {
+    if (!this.getDefaultAddress()) {
+      throw new InternalError("Default address not found");
+    }
+    return await this.getDefaultAddress()!.claimableBalance(asset_id, mode, options);
+  }
+
+  /**
+   * Lists the staking rewards for the address.
+   *
+   * @param assetId - The asset ID.
+   * @param startTime - The start time.
+   * @param endTime - The end time.
+   * @param format - The format to return the rewards in. (usd, native). Defaults to usd.
+   * @throws {Error} if the default address is not found.
+   * @returns The staking rewards.
+   */
+  public async stakingRewards(
+    assetId: string,
+    startTime = getWeekBackDate(new Date()),
+    endTime = formatDate(new Date()),
+    format: StakingRewardFormat = StakingRewardFormat.Usd,
+  ): Promise<StakingReward[]> {
+    if (!this.getDefaultAddress()) {
+      throw new InternalError("Default address not found");
+    }
+    return await this.getDefaultAddress()!.stakingRewards(assetId, startTime, endTime, format);
+  }
+
+  /**
+   * Creates a staking operation to stake, signs it, and broadcasts it on the blockchain.
+   *
+   * @param amount - The amount for the staking operation.
+   * @param assetId - The asset for the staking operation.
+   * @param action - The type of staking action to perform.
+   * @param timeoutSeconds - The amount to wait for the transaction to complete when broadcasted.
+   * @param intervalSeconds - The amount to check each time for a successful broadcast.
+   * @param options - Additional options such as setting the mode for the staking action.
+   * @throws {Error} if the default address is not found.
+   * @returns The staking operation after it's completed fully.
+   */
+  public async createStakingOperation(
+    amount: Amount,
+    assetId: string,
+    action: string,
+    timeoutSeconds = 60,
+    intervalSeconds = 0.2,
+    options: CoinbaseWalletAddressStakeOptions = { mode: StakeOptionsMode.DEFAULT },
```

Comment:
this func takes amount as input, and `CoinbaseWalletAddressStakeOptions` also has amount in it.
We should instead take mode as an explicit input and options separatey

Reply:
Also looks like we will take the route of renaming this to `createStakeOperation` ? and add the other 2 as well ?

Reply:
Yeah I added the refactor a few moments ago. They will be called `createStake` , `createUnstake` , and `createClaimStake`.

So based on your first comment, is this how you're suggesting the main function signature to look:
```
public async createStake(
    amount: Amount,
    assetId: string,
    mode = StakeOptionsMode.DEFAULT,
    timeoutSeconds = 60,
    intervalSeconds = 0.2,
    options: CoinbaseWalletAddressStakeOptions = {},
  ): Promise<StakingOperation> 
```
@drohit-cb

Reply:
mode and options will look like this https://github.com/coinbase/coinbase-sdk-nodejs/blob/master/src/coinbase/address/external_address.ts#L32-L33

Reply:
and the order of params can be `amount, assetID, mode, options, timeoutSeconds, intervalSeconds`

Reply:
refactored based on the suggestion.